services/data_deletion/clean_archived_data.py

```diff
@@ -0,0 +1,22 @@
+from helpers.database import database_uri
+from sqlalchemy import create_engine, MetaData, and_
+from datetime import timedelta, datetime
+import celery
+
+
+@celery.task(name='clean_archived_data.delete_archived_data')
+def delete_archived_data():
+    database_engine = create_engine(database_uri)
+    metadata = MetaData()
+    metadata.reflect(bind=database_engine)
+    print('deleting archived data...')
+
+    for table in reversed(metadata.sorted_tables):
+        try:
+            now = datetime.now()
+            delta = now - timedelta(days=30)
+            statement = table.delete().where(
+                and_(table.c.date_updated < delta, table.c.state == 'archived'))
```

Comment:
@mifeille after finishing the job, this is what celery.err.log is saying `sqlalchemy.exc.IntegrityError: (psycopg2.IntegrityError) update or delete on table "rooms" violates foreign key constraint "room_tags_room_id_fkey" on table "room_tags"
DETAIL:  Key (id)=(1) is still referenced from table "room_tags".
 [SQL: 'DELETE FROM rooms WHERE rooms.date_updated < %(date_updated_1)s AND rooms.state = %(state_1)s'] [parameters: {'date_updated_1': datetime.datetime(2019, 10, 5, 14, 31, 1, 378610), 'state_1': 'archived'}] (Background on this error at: http://sqlalche.me/e/gkpj)`

Reply:
There is a migration to include cascading deletion I guess that one on room tags was forgotten, I am going to add it.